src/Resources/config/verify_email_services.xml

```diff
@@ -11,16 +11,23 @@
 
         <service id="symfonycasts.verify_email.query_utility" class="SymfonyCasts\Bundle\VerifyEmail\Util\VerifyEmailQueryUtility" public="false" />
 
+        <!-- TODO Fix this service for BC ?
         <service id="symfonycasts.verify_email.uri_signer" class="Symfony\Component\HttpKernel\UriSigner">
             <argument>%kernel.secret%</argument>
             <argument>signature</argument>
         </service>
```

Comment:
Should we worry about registering this service for BC? We don't need it anymore because we use `symfonycasts.verify_email.uri_signer_factory` now but probably some users may use it?

P.s. not sure how we can register this service for different available `UriSigner` classes

Reply:
Ok, what we can do is this:

A) Keep this service. But make it a "factory" service. So, make it CALL the new `symfonycasts.verify_email.uri_signer_factory` to get created https://symfony.com/doc/current/service_container/factories.html

B) Then, the argument to `symfonycasts.verify_email.helper` can be reverted: we can go back to injecting the original service.

Reply:
@weaverryan I did A. About B - I don't think it's possible, we need a typehint... but new and legacy UrlSigner classes have different namespaces, so I can't autocomplete both I suppose. It seems like using Factory typeing is the correct way, or am I missing something? Wdyt?

Reply:
We should still pass in a `UriSigner` instance (not a factory)... just because all we need is the instance (the fact that we're using a factory to create it is a DI detail).

About the type-hint on the argument, if our minimum PHP version were 8.0 (and yes, we should bump to at least that - probably 8.1), then we could use a union type - `UriSigner|LegacyUriSigner`. Or, to make life simpler for this PR (and push updating PHP until later), we could remove the type-hint. So we *would* pass in a `UriSigner` instance, but we wouldn't have an actual type-hint on that argument.